Modules/Brand/Http/Services/BrandService.php

```diff
@@ -0,0 +1,287 @@
+    {
+
+        $brandData["brand_key"] = 'bmc_' . Str::lower(Str::random(10));
+        $slug = Str::slug($brandData["brand_name"], '-');
+        $count = Brand::where(DB::raw('lower(brand_name)'), strtolower($brandData["brand_name"]))->count();
+        if ($count > 0) {
+            $slug = $slug . '-' . ($count+1);
+        }
+        $password = $brandData['password'];
+        $brandData["brand_slug"] = $slug;
+        $brandData["bazaar_direct_link"] = $slug;
+
+        $document = $request->file('document');
+        $tempPath = $document->store('temp', 'public');
+        $destinationPath = '../public/uploads/documents/';
+        $newFileName = time() . '_' . $document->getClientOriginalName();
+        $document->move($destinationPath, $newFileName);
+        \Storage::disk('public')->delete($tempPath);
+
+        $user = new User();
+        $user->first_name = $brandData["first_name"];
+        $user->last_name = $brandData["last_name"];
+        $user->email = $brandData["email"];
+        $user->password = Hash::make($brandData['password']);
+        $user->role = ROLE::ROLE_BRAND;
+        $user->save();
+
+        $brand = new Brand();
+        $brand->user_id = $user->id;
+        $brand->brand_key = $brandData["brand_key"];
+        $brand->brand_email = $brandData["brand_email"];
+        $brand->brand_name = $brandData["brand_name"];
+        $brand->brand_slug  = $brandData["brand_slug"];
+        $brand->bazaar_direct_link = $brandData["bazaar_direct_link"];
+        $brand->about_us = $brandData["about_us"];
+        $brand->num_store = $brandData["num_store"];
+        $brand->prime_cat = $brandData["prime_cat"];
+        $brand->website_url = $brandData["website_url"];
+        $brand->country_code = $brandData["country_code"];
+        $brand->country = $brandData["country"];
+        $brand->phone_number = $brandData["phone_number"];
+        $brand->language = $brandData["language"];
+        $brand->headquatered = $brandData["headquatered"];
+        $brand->established_year = $brandData["established_year"];
+        $brand->insta_handle = $brandData["insta_handle"];
+        $brand->city = $brandData["city"];
+        $brand->state = $brandData["state"];
+        $brand->product_made = $brandData["product_made"];
+        $brand->product_shipped = $brandData["product_shipped"];
+        $brand->num_products_sell = $brandData["num_products_sell"];
+        $brand->document = 'uploads/documents/'.$newFileName;
+
+
+        $brand->save();
+
+        $webUrl = config('app.web_url');
+        Mail::send('brand_email', ['webUrl' => $webUrl, 'first_name' => $brandData["first_name"],'email' => $brandData["email"],'password' => $password], function ($message) use ($brandData) {
+            $message->to($brandData['email']);
+            $message->from(config('app.from_email'));
+            $message->subject('Bazar:Your account has been created');
+        });
```

Comment:
Try to make a separate function for sending mail, in case we need to send mail again somewhere, you can directly use that function. That's the benefit of OOP.

Reply:
Fixed